### DISCUSSION TITLE
Custom fields for submissions

### DISCUSSION THREAD

**Discussion by kaennchenstruggle** [pretalx/pretalx#1276] [Ideas] [answered]
Especially online events see the requirement for a field to announce a link to a virtual room (BBB/Jitsi/...) that may be different for each talk. To add some abstraction, I would propose to enable custom fields per submission.

Field flavors could be 
- text
- link
- markdown

Field properties could be:
- visibility: group, pretalx, everybody
- modify: group, author, system

Plus an API endpoint where the contents of the field can be automagically modified (by script)

**Comment by rixx** [comment, marked as answer]
I believe you're looking for the questions feature, which allows organisers and speakers to add information to talks.

Questions

- can be per submission or per speakers
- have different field flavours
- can be publicly visible or only to speakers and organisers or also to reviewers
- have an API and an export

**Reply by kaennchenstruggle** [reply]
Agreed, this feature could be used to achieve the goal. But how can we change data in this field automatically (administratively, from outside, by script)

**Reply by rixx** [reply]
I'm not sure I understand what you're asking – as I've said, questions have an API. You can create and update question answers by using the API. Is there anything else you'd need?

**Reply by kaennchenstruggle** [reply]
I read the API as an ability to GET data from pretalx but didn't find an option to POST an update to the question (answer). Perhaps I read it the wrong way. Or is there a bulk import feature for answers to questions?

**Reply by rixx** [reply]
There is no bulk feature. You have to make individual requests per question.

**Reply by kaennchenstruggle** [reply]
found out, how to post data to the answers list. thanks for your help! And sorry for the misunderstandings
